### ISSUE TITLE
Add a card on STPPaymentOptionsViewController doesn't work for the first time.

### ISSUE THREAD

**Issue by daigok** [stripe/stripe-ios#1758]
## Summary
I'm using STPPaymentOptionsViewController with SwiftUI. The ViewController shows "Add a Card" view if a customer doesn't have a card. If I try to add a card on the "Add a Card" view, it keep loading forever and doesn't call registering apis.

 It worked fine with SDK 21.0.1 or lower.

## Code to reproduce
The code snippet  I'm using is [here](https://github.com/stripe/stripe-ios/issues/1601#issuecomment-772312400).

## iOS version
14.4

## Installation method
Swift Package Manager

## SDK version
21.2.1

## Other information
I've debugged the SDK and found apiAdapter on STPPaymentOptionsViewController is nil for the first card addition.
https://github.com/stripe/stripe-ios/blob/21.2.1/Stripe/STPPaymentOptionsViewController.swift#L278

This is because apiAdapter is not used if a customer doesn't have a card on the view controller initialization and also apiAdapter is defined as weak.
https://github.com/stripe/stripe-ios/blob/21.2.1/Stripe/STPPaymentOptionsViewController.swift#L88-L127

Hence, the following code is not called and the view keeps loading forever. 
https://github.com/stripe/stripe-ios/blob/21.2.1/Stripe/STPPaymentOptionsViewController.swift#L437

Add a Card view keeps loading forever. 
![Simulator Screen Shot - iPhone SE (2nd generation) - 2021-02-04 at 17 12 51](https://user-images.githubusercontent.com/4699741/106863729-446a0780-670c-11eb-996f-3def6adf28e9.png)

**Comment by Ariandr** [comment]
I have the same issue on 21.2.1 (not using SwiftUI)

Happens when I initialize `STPPaymentOptionsViewController ` using this initializer:
```
  public convenience init(
    configuration: STPPaymentConfiguration,
    theme: STPTheme,
    customerContext: STPCustomerContext,
    delegate: STPPaymentOptionsViewControllerDelegate
  )
```

I was doing it for the purpose of managing payment options in a "Payment methods" menu item in the user settings in the app.
So, my behavior required not closing the controller after pressing on a method (or adding new card), but rather setting it as a default.

Also, when I present the controller by calling on payment context (present or push Payment Options Controller), it works for the first card.

**Comment by csabol-stripe** [comment]
@daigok thanks for the detailed report! I've got a fix for this that will be included in the next release :)

**Comment by davidme-stripe** [comment]
This is fixed in [21.3.0](https://github.com/stripe/stripe-ios/releases/tag/21.3.0).
